src/index.js

```diff
@@ -7,17 +7,22 @@ export class PhoenixLiveViewDropzone {
     var node = document.createElement("div");
     this.el.appendChild(node);
 
-    var liveView = this;
+    this.target = this.el.dataset.target;
 
-    var generateUrl, statusUpdate;
+    this.generateUrl = (data) => {
+      if (this.target) {
+        this.pushEventTo(this.target, "phx-dropzone", ["generate-url", data]);
+      } else {
+        this.pushEvent("phx-dropzone", ["generate-url", data]);
+      }
+    }
 
-    if (this.el.dataset.target) {
-      var target = this.el.dataset.target;
-      generateUrl = (data) => liveView.pushEventTo(target, "phx-dropzone", ["generate-url", data]);
-      statusUpdate = (data) => liveView.pushEventTo(target, "phx-dropzone", ["file-status", data]);
```

Comment:
I think this is the source of the issue, the closures here have captured the current version of target, where as if it was  `liveview.el.dataset.target` it would be the current target value

Reply:
True, though it would still be an issue if a component ever switched between having a target defined and not. I don't think that's a huge problem, it wouldn't affect me, but I think it makes sense to do the check within the callback as opposed to when we are defining the callback.

Reply:
Can you see if #9 fixes your problem? I've checked it still works for me but I'm not rebinding target...

Reply:
#9 _does_ fix my problem. Thanks!